ipfs-api-prelude/src/backend.rs

```diff
@@ -236,4 +236,7 @@ pub trait Backend {
             )
         })
     }
+
+    /// Set basic authentication credentials to use on every request from this client.
+    fn with_credentials(self, username: &str, password: &str) -> Self;
```

Comment:
Thoughts about making this something like: 

```
fn with_credentials(self, username: impl Into<String>, password: Into<String>) -> Self;
```

for some more flexibility?

Reply:
This is a good idea. I was waffling between `Into<String>` and `Borrow<str>` so left it `&str`